gammapy/spectrum/models.py

```diff
@@ -519,11 +518,14 @@ class PowerLaw(SpectralModel):
     """
 
     def __init__(self, index=2., amplitude=1E-12 * u.Unit('cm-2 s-1 TeV-1'),
-                 reference=1 * u.TeV):
+                 reference=1 * u.TeV, name=None):
+        if name is None:
+            name = self.__class__.__name__
```

Comment:
Maybe make it one line, since we'll get many of those?
```
self.name = name or self.__class__.__name__
```
Is it possible to just put this line in the base class?